src/AzureExtension/DevBox/Helpers/WingetConfigWrapper.cs

```diff
@@ -328,6 +328,51 @@ private void SetStateForCustomizationTask(TaskJSONToCSClasses.BaseClass response
         }
     }
 
+    // Start the Dev Box if it isn't already running
+    private async Task HandleNonRunningState()
+    {
+        // Check if the dev box might have been started in the meantime
+        var stateRequest = await _managementService.HttpsRequestToDataPlane(new Uri(_baseAPI), _devId, HttpMethod.Get, null);
+        var response = JsonSerializer.Deserialize<DevBoxMachineState>(stateRequest.JsonResponseRoot.ToString(), Constants.JsonOptions)!;
+
+        if (response.PowerState == Constants.DevBoxPowerStates.Running)
+        {
+            return;
+        }
+
+        // Start the Dev Box
+        var startURI = new Uri($"{_baseAPI}:start?{Constants.APIVersion}");
+        await _managementService.HttpsRequestToDataPlane(startURI, _devId, HttpMethod.Post, null);
+        _log.Information("Starting the dev box to apply configuration");
+
+        // Notify the user
+        ConfigurationSetStateChanged?.Invoke(this, new(new(ConfigurationSetChangeEventType.SetStateChanged, ConfigurationSetState.StartingDevice, ConfigurationUnitState.Unknown, null, null)));
+
+        // Wait for the Dev Box to start with an initial wait of 3 minutes
+        var startState = string.Empty;
+        await Task.Delay(TimeSpan.FromMinutes(3));
+
+        // Keep polling for another 12 minutes
+        var tries = 0;
+        while (tries++ <= 24 && startState != Constants.DevBoxPowerStates.Running)
+        {
+            await Task.Delay(TimeSpan.FromSeconds(30));
+            var poll = await _managementService.HttpsRequestToDataPlane(new Uri(_baseAPI), _devId, HttpMethod.Get, null);
+            var pollingResponse = JsonSerializer.Deserialize<DevBoxMachineState>(poll.JsonResponseRoot.ToString(), Constants.JsonOptions)!;
+            startState = pollingResponse.PowerState;
+        }
+
+        // If there was a timeout
+        if (tries == 25)

```

Comment:
There are 3 constants here and changing one of them requires changing others and/or the comment about 12 minutes. (24, 25 for attempts, and 30 for delay)
Consider simplifying this as only the total time (12 minutes) and delay are needed.
Something like below would state maximum waiting time and delay between attempts in the code (so no need to do math with the attempts counter):
```
        var delay = TimeSpan.FromSeconds(30);
        var waitTimeLeft = TimeSpan.FromMinutes(12);
        while ((waitTimeLeft > TimeSpan.Zero) && (startState != Constants.DevBoxPowerStates.Running))
        {
            await Task.Delay(delay);
            var poll = await _managementService.HttpsRequestToDataPlane(new Uri(_baseAPI), _devId, HttpMethod.Get, null);
            var pollingResponse = JsonSerializer.Deserialize<DevBoxMachineState>(poll.JsonResponseRoot.ToString(), Constants.JsonOptions)!;
            startState = pollingResponse.PowerState;
            waitTimeLeft -= delay;
        }

        // If there was a timeout
        if (startState != Constants.DevBoxPowerStates.Running)
        {
            throw new InvalidOperationException(Resources.GetResource(DevBoxErrorStateKey));
        }
        else
        {
            _log.Information($"Started successfully after {tries} tries");
        }
```

Checking for max number of attempts is not needed as you can check for the state instead.

Reply:
Changed